esi_leap/api/controllers/v1/utils.py

```diff
@@ -0,0 +1,130 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_policy import policy as oslo_policy
+from oslo_utils import uuidutils
+
+from esi_leap.common import exception
+from esi_leap.common import policy
+from esi_leap.common import statuses
+from esi_leap.objects import contract
+from esi_leap.objects import offer
+from esi_leap.resource_objects import resource_object_factory as ro_factory
+
+
+def get_authorized_offer(uuid_or_name, status_filter, cdict):
+    if uuidutils.is_uuid_like(uuid_or_name):
+        o = offer.Offer.get(uuid_or_name)
+        offer_objs = []
+
+        if (status_filter and o.status == status_filter) or \
```

Comment:
If you make status_filter an optional parameter that defaults to [], this can just be `if not status_filter or o.status in status_filter`

Reply:
I assume there will only be one status passed in for the offers, whereas in the contracts I assume there can be multiple passed in. I do this so I don't have to change the dbapi for this, and we currently do not have a use case where there would be multiple statuses that we want to get

Reply:
Ah - then this could just be: `if not status_filter or o.status == status_filter`.

Reply:
fixed